pyrdp/security/nla.py

```diff
@@ -24,25 +24,30 @@ class NLAHandler(SegmentationObserver):
     This also logs the hash of NLA connection attempts.
     """
 
-    def __init__(self, sink: IntermediateLayer, state: NTLMSSPState, log: logging.LoggerAdapter, ntlmCapture: bool = False):
+    def __init__(self, sink: IntermediateLayer, state: NTLMSSPState, log: logging.LoggerAdapter, ntlmCapture: bool = False, challenge: str = None):
         """
         Create a new NLA Handler.
         sink: layer to forward packets to.
         state: NTLMSSPState that is shared between both the client-facing handler and the server-facing handler.
         """
+        print("init calld")
 
         super().__init__()
         self.sink = sink
         self.ntlmSSPState = state
         self.ntlmSSPParser = NTLMSSPParser()
         self.ntlmCapture = ntlmCapture
+        self.challenge = challenge
         self.log = log
 
-    def getRandChallenge(self):
+    def getChallenge(self):
         """
-        Generate a random 64-bit challenge
+        Return configured challenge or a random 64-bit challenge
         """
-        challenge = b'%016x' % secrets.randbits(16 * 4)
+        if self.challenge == None:
```

Comment:
I believe this `None` check should use `is`

```suggestion
        if self.challenge is None:
```